extensions/amp-subscriptions/0.1/entitlement-store.js

```diff
@@ -101,10 +101,65 @@ export class EntitlementStore {
 
   /**
    * Returns entitlements when all services are done fetching them.
-   * @returns {!Promise<Entitlements>}
+   * @private
+   * @returns {!Promise<!Array<!Entitlements>>}
    */
-  getAllPlatformsEntitlement() {
-    // TODO(@prateekbh): implement this.
+  getAllPlatformsEntitlements_() {
+    if (this.allResolvedPromise_) {
+      return this.allResolvedPromise_;
+    }
+
+    this.allResolvedPromise_ = new Promise(resolve => {
+      if (this.areAllPlatformsResolved_()) {
+        // Resolve with null if non of the entitlements unblocks the reader
+        return resolve(Object.values(this.entitlements_));
+      } else {
+        // Listen if any upcoming entitlements unblock the reader
+        this.onChange(() => {
+          if (this.areAllPlatformsResolved_()) {
+            resolve(Object.values(this.entitlements_));
+          }
+        });
+      }
+    });
+
+    return this.allResolvedPromise_;
+  }
+
+  /**
+   * Returns entitlements when all services are done fetching them.
+   * @returns {!Promise<!Entitlements>}
+   */
+  selectPlatform() {
+    return this.getAllPlatformsEntitlements_().then(entitlements => {
+      return this.selectApplicablePlatform_(entitlements);
```

Comment:
Add a `TODO` to explain that in the future we may be able to resolve a selected platform faster than waiting for all platforms. E.g. a first "subscriber" platform can be selected right away.

Reply:
Done